app/models/event.rb

```diff
@@ -44,10 +46,17 @@ def pinned=(val)
     end
   end
 
+  def normalize_friendly_id(string)
+    PinYin.permlink(header).downcase
+  end
+
   private
 
   def end_time_cannot_before_start_time
     errors.add(:end_at, "不能早于开始时间") if end_at < start_at
   end
 
+  def should_generate_new_friendly_id?
```

Comment:
this is a special hook provided by friendly-id gem, should add a comment for this method